### PR TITLE
Notification remove use flags

### PR DESCRIPTION
**Pull request by difince**
- delete by age and slug happens by using flags
  - change Usage msg, and flags description

Signed-off-by: Diana Atanasova <dianaa@vmware.com>

### CODE REVIEW

File: cmd/notification/rm/rm.go [hunk 1]
```diff
@@ -24,32 +26,36 @@ import (
...
 
+// NewCommand returns the rm command of type cobra.Command
 func NewCommand() *cobra.Command {
 	var cmd = &cobra.Command{
-		Use:   "rm [notification slug or age]",
+		Use:   "rm [--slug | --age]",
 		Short: "Removes notification by slug or age",
-		Long:  `Removes a notification given its slug or age timestamp.`,
-		Args:  cobra.ExactValidArgs(1),
+		Long:  "Removes notifications by slug or age (the current notifications timestamp minus their last modification timestamp should be less than the age parameter)",
```
Comment (mkbhanda) [inline cmd/notification/rm/rm.go:47]: s/less/greater ??
Say age is 5 hours, anything greater than 5 hours, like 5 hrs and 1 minute, 6 hrs, 24 hrs should get deleted? With notifications are we looking at when we it was created and current time?

File: cmd/notification/rm/rm.go [hunk 2]
```diff
@@ -24,32 +26,36 @@ import (
...
 		RunE:   removeNotificationHandler,
 	}
-	cmd.Flags().BoolVar(&byAge, "age", false, "Remove by age")
+	cmd.Flags().StringVar(&age, "age", "", "Notification age (in milliseconds)")
```
Comment (mkbhanda) [inline cmd/notification/rm/rm.go:50]: does it make sense to change milliseconds to minutes? what is the most likely use case?
Might someone want to delete all notifications from yesterday or an hour back? translating to milli seconds may be a pain?

File: cmd/notification/rm/rm.go [hunk 3]
```diff
@@ -24,32 +26,36 @@ import (
...
 	return cmd
 }
+
+func constructUrl() (string, string) {
+	url := config.Conf.Clients["Notification"].Url() + clients.ApiNotificationRoute
+	if age != "" {
+		return fmt.Sprintf("%s/age/%s", url, age), age
+	}
+	return fmt.Sprintf("%s/slug/%s", url, slug), slug
+}
```
Comment (mkbhanda) [inline cmd/notification/rm/rm.go:61]: might want to add a new line to get rid of this warning

File: cmd/notification/rm/rm.go (hunk 1 above)
Comment (difince) [reply cmd/notification/rm/rm.go:47]: I took the description out of the api v1 documentation. 
The logic implemented is: 
curr_time - 5 hours  =  time in the past (5 hours earlier) (all in ms) 
If this past_time is less than the last modification time - the notification will be deleted  

But said in this way it should be really greater 
> the current notifications timestamp minus their last modification timestamp should be less than the age parameter

File: cmd/notification/rm/rm.go (hunk 2 above)
Comment (difince) [reply cmd/notification/rm/rm.go:50]: I have added another optional flag called --unit that could accept time units  ms, s , m , h, d ... 
The default value of this flag is ms (milliseconds) 

Examples delete by age: 
./edgex-cli notification rm --age 1 --unit h  (for 1 hour)
./edgex-cli notification rm --age 45 --unit m  (for 45 minutes)
./edgex-cli notification rm --age 3 --unit d  (for  3 days)

Example remove notification by slug:
./edgex-cli notification rm --slug slug_name

**Comment by difince** [comment]
FYI, I have tested delete notification by Age ,with Mongo database only. There is a bug inside the mongo implementation that makes this requests doing nothing. 
https://github.com/edgexfoundry/edgex-go/blob/07eb6246d5fb2c01df3c6ba4777e936c65ab9351/internal/pkg/db/mongo/notifications.go#L189 
It looks that  `currentMaxResultCount`  is legacy, and its value is always 0.. this make the select request (get notification by age to be limited to 0 result - select nothing) I am not going to open issue about that because Mongo is already deprecated and the code is about to be removed soon and we are not going to support it , fix issues within the older versions

**Comment by mkbhanda** [comment]
Nice work Diana! It may still be worth tossing in a bug on the Mongo limit 0 issue even if you mark it as "will not fix, removing MongoDB" because it affects existing releases. Some developer will not need to debug the issue if they encounter it and save time.
I also think we should put a doc bug for the the age greater text in edgex-go.

**Pull request merged**